### PR TITLE
enh(resources): migrate listing endpoint to clean architecture

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

This PR intends to migrate Resource Status listing endpoint to the new API architecture.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

See Jira ticket

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Resources/Application/UseCase/FindResources/Response/ResourceResponseDto.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        public ?int $hostId = null,
+        public ?string $alias = null,
+        public ?string $fqdn = null,
+        public ?IconResponseDto $icon = null,
+        public ?ResourceStatusResponseDto $status = null,
+        public bool $isInDowntime = false,
+        public bool $isAcknowledged = false,
+        public bool $withActiveChecks = false,
+        public bool $withPassiveChecks = false,
+        public ?\DateTime $lastStatusChange = null,
```
Comment (adr-mo) [inline centreon/src/Core/Resources/Application/UseCase/FindResources/Response/ResourceResponseDto.php:70]: Check for DateTimeImmutable

File: centreon/src/Core/Resources/Infrastructure/API/FindResources/FindResourcesRoute.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FindResources:
+    methods: GET
```
Comment (adr-mo) [inline centreon/src/Core/Resources/Infrastructure/API/FindResources/FindResourcesRoute.yaml:2]: delete old definition of route

**Review by kduret — COMMENTED** [review]
I think you need to remove those lines : https://github.com/centreon/centreon/blob/MON-20898/centreon/config/packages/Centreon.yaml#L7-L9

**Comment by adr-mo** [comment]
@kduret thanks I've removed those lines :)

**Review by kduret — APPROVED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by callapa — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**